org-eclipse-che-jdt-ls-extension/jdt.ls.extension.core/src/main/java/org/eclipse/che/jdt/ls/extension/core/internal/navigation/FindImplementorsHandler.java

```diff
@@ -0,0 +1,163 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.jdt.ls.extension.core.internal.navigation;
+
+import static java.util.Collections.emptyList;
+import static org.eclipse.che.jdt.ls.extension.core.internal.JavaModelUtil.getJavaProject;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import java.util.ArrayList;
+import java.util.List;
+import org.eclipse.che.jdt.ls.extension.api.dto.navigation.FindImplementationsCommandParameters;
+import org.eclipse.che.jdt.ls.extension.api.dto.navigation.ImplementationsDescriptor;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.ICodeAssist;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.IMethod;
+import org.eclipse.jdt.core.IType;
+import org.eclipse.jdt.core.ITypeHierarchy;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.jdt.internal.corext.util.MethodOverrideTester;
+import org.eclipse.jdt.ls.core.internal.JDTUtils;
+import org.eclipse.jdt.ls.core.internal.ResourceUtils;
+import org.eclipse.jdt.ls.core.internal.handlers.DocumentSymbolHandler;
+import org.eclipse.jdt.ls.core.internal.hover.JavaElementLabels;
+import org.eclipse.lsp4j.Location;
+import org.eclipse.lsp4j.SymbolInformation;
+import org.eclipse.lsp4j.jsonrpc.json.adapters.CollectionTypeAdapterFactory;
+import org.eclipse.lsp4j.jsonrpc.json.adapters.EitherTypeAdapterFactory;
+import org.eclipse.lsp4j.jsonrpc.json.adapters.EnumTypeAdapterFactory;
+
+public class FindImplementorsHandler {
+  private static final Gson gson =
+      new GsonBuilder()
+          .registerTypeAdapterFactory(new CollectionTypeAdapterFactory())
+          .registerTypeAdapterFactory(new EitherTypeAdapterFactory())
+          .registerTypeAdapterFactory(new EnumTypeAdapterFactory())
+          .create();
+
+  public static Object getImplementors(List<Object> parameters, IProgressMonitor pm) {
+    FindImplementationsCommandParameters commandParameters =
+        gson.fromJson(gson.toJson(parameters.get(0)), FindImplementationsCommandParameters.class);
+
+    IJavaProject javaProject = getJavaProject(commandParameters.getProjectUri());
+
+    if (javaProject == null) {
+      return emptyList();
+    }
+
+    ImplementationsDescriptor implementationsDescriptor = new ImplementationsDescriptor();
+
+    try {
+      IJavaElement element =
+          getJavaElement(javaProject, commandParameters.getFqn(), commandParameters.getOffset());
+      if (element == null) {
+        implementationsDescriptor.setImplementations(emptyList());
+        return implementationsDescriptor;
+      }
+
+      List<SymbolInformation> implementations = new ArrayList<>();
+
+      implementationsDescriptor.setImplementations(implementations);
+
+      switch (element.getElementType()) {
+        case 7: // type
```

Comment:
Are there any constant that we can use?

Reply:
I would even say we _should_ use: See IJavaElement.